src/legacy/core_plugins/kibana/public/management/sections/index_patterns/create_index_pattern_wizard/create_index_pattern_wizard.test.tsx

```diff
@@ -167,36 +143,29 @@ describe('CreateIndexPatternWizard', () => {
   });
 
   it('invokes the provided services when creating an index pattern', async () => {
-    const get = jest.fn();
-    const set = jest.fn();
     const create = jest.fn().mockImplementation(() => 'id');
     const clear = jest.fn();
-    const changeUrl = jest.fn();
+    services.indexPatterns.clearCache = clear;
+    const indexPattern = ({
+      id: '1',
+      title: 'my-fake-index-pattern',
+      timeFieldName: 'timestamp',
+      fields: [],
+      create,
+    } as unknown) as IndexPattern;
+    services.indexPatterns.make = async () => {
+      return indexPattern;
+    };
 
     const component = shallow(
-      <CreateIndexPatternWizard
-        loadingDataDocUrl={loadingDataDocUrl}
-        initialQuery={initialQuery}
-        services={{
-          ...services,
-          config: { get, set },
-          indexPatterns: {
-            make: () => ({
-              create,
-            }),
-            clearCache: clear,
-          },
-          changeUrl,
-          indexPatternCreationType: mockIndexPatternCreationType,
-        }}
-      />
+      <CreateIndexPatternWizard initialQuery={initialQuery} services={services} />
     );
 
     component.setState({ indexPattern: 'foo' });
-    await component.instance().createIndexPattern(null, 'id');
-    expect(get).toBeCalled();
+    await (component.instance() as CreateIndexPatternWizard).createIndexPattern(undefined, 'id');
```

Comment:
Let's use generics like e.g. shallow<CreateIndexPatternWizard> instead of casting to CreateIndexPatternWizard

Reply:
done